share/sysbuild/image_configurations/MAIN_image_default.cmake

```diff
@@ -37,4 +37,16 @@ if(SB_CONFIG_BOOTLOADER_MCUBOOT)
   elseif(SB_CONFIG_MCUBOOT_MODE_FIRMWARE_UPDATER)
     set_config_bool(${ZCMAKE_APPLICATION} CONFIG_MCUBOOT_BOOTLOADER_MODE_FIRMWARE_UPDATER y)
   endif()
+
+  set(imgtool_extra_args "${CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS}")
+  if(SB_CONFIG_BOOT_SIGNATURE_TYPE_ED25519)
+    if(SB_CONFIG_BOOT_SIGNATURE_TYPE_PURE)
+      string(APPEND imgtool_extra_args " --pure")
+    else()
+      string(APPEND imgtool_extra_args " --sha 512")
+    endif()
+  endif()
+
```

Comment:
this should not be in sdk-zephyr at all, move it to sdk-nrf, as part of sysbuild, and set Kconfigs in the image which then enable these options, do not just the extra args - that field is for users to change as they want